src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -80,7 +81,19 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
         }
 
         if (this.props.selectionMode !== prevProps.selectionMode) {
-            this.editor.AS.setSelectionMode(this.props.selectionMode, null);
+            let  options = null;
+            if (this.props.selectionMode === SelectionMode.COPYRECT) {
+                const selectedRegion = this.getSelectedRegions();
```

Comment:
Doesn't this return an array of regions?  If no selected regions does it return null or array of zero?  Please confirm and update logic as needed.

Also - I feel that the `componentDidUpdate` is not the place to specify the template options.  I would rather see the template selection happen on region selected or on draw of the region